### PR TITLE
Create schema-info for global schema info

### PR DESCRIPTION
**Pull request by bratseth**
@arnej27959 Lester is away so perhaps you could do this one?

This is just creating a new config to hold schema info, the same as documentdb-info, but available independently of search clusters.

I plan to simplify by switching everything over to use it and get rid of documentdb-info since we don't actually need this to be different per cluster. But I don't want to do all in one PR.

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/searchchain/Execution.java [hunk 1]
```diff
@@ -140,7 +140,7 @@ public Context(SearchChainRegistry searchChainRegistry, IndexFacts indexFacts, S
         public Context(SearchChainRegistry searchChainRegistry, IndexFacts indexFacts,
                        SpecialTokenRegistry tokenRegistry, RendererRegistry rendererRegistry, Linguistics linguistics,
                        Executor executor) {
-            this(searchChainRegistry, indexFacts, SchemaInfo.empty(), tokenRegistry, rendererRegistry, linguistics, Runnable::run);
+            this(searchChainRegistry, indexFacts, SchemaInfo.empty(), tokenRegistry, rendererRegistry, linguistics, executor);
```
Comment (arnej27959) [inline container-search/src/main/java/com/yahoo/search/searchchain/Execution.java:142]: this change looks totally unrelated

File: container-search/src/main/resources/configdefinitions/container.search.schema-info.def [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+namespace=container.search
+
+## The name of this schema
+schema[].name string
+
+## The id of the summary class
```
Comment (arnej27959) [inline container-search/src/main/resources/configdefinitions/container.search.schema-info.def:7]: this id isn't used anymore, can be dropped

File: container-search/src/main/resources/configdefinitions/container.search.schema-info.def [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# schema[].summaryclass[].fields[].name string
+## The type of a field in the summary class
+# schema[].summaryclass[].fields[].type string
+## Whether this field is a dynamic snippet
+# schema[].summaryclass[].fields[].dynamic bool default=false
+
+## Information about rank profiles
+schema[].rankprofile[].name string
+schema[].rankprofile[].hasSummaryFeatures bool default=true
+schema[].rankprofile[].hasRankFeatures bool default=true
```
Comment (arnej27959) [inline container-search/src/main/resources/configdefinitions/container.search.schema-info.def:21]: i can't find any place where hasRankFeatures is actually used.

**Review by arnej27959 — APPROVED** [review]
looks good to me, only minor comments

File: container-search/src/main/java/com/yahoo/search/searchchain/Execution.java (hunk 1 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/searchchain/Execution.java:142]: It fixes a bug in an unused method preserved for compatibility, introduced in an earlier part of this work.

File: container-search/src/main/resources/configdefinitions/container.search.schema-info.def (hunk 2 above)
Comment (bratseth) [reply container-search/src/main/resources/configdefinitions/container.search.schema-info.def:7]: Yes, it is commented out. I'll be adding it back when I get rid of documentdb-config, but for that I need to rewrite the way the summary part of that config is generated as it currently is generated from another config, which is no-no.

File: container-search/src/main/resources/configdefinitions/container.search.schema-info.def (hunk 3 above)
Comment (bratseth) [reply container-search/src/main/resources/configdefinitions/container.search.schema-info.def:21]: This was easier to keep. It will be used when I move away from documentdb-info.
